src/schema.rs

```diff
@@ -352,10 +378,115 @@ fn overwrite_toml_output(
     Ok(())
 }
 
+fn overwrite_toml_status(status: ExitStatus, raw: String) -> Result<String, toml_edit::TomlError> {
+    let mut doc = raw.parse::<toml_edit::Document>()?;
+    if let Some(code) = status.code() {
+        if status.success() {
+            if match doc.get("status") {
+                Some(toml_edit::Item::Value(toml_edit::Value::String(ref expected)))
+                    if expected.value() == "success" =>
+                {
+                    false
+                }
+                Some(
+                    toml_edit::Item::Value(toml_edit::Value::InlineTable(_))
+                    | toml_edit::Item::Table(_),
+                ) => !matches!(
+                    doc["status"].get("code"),
+                    Some(toml_edit::Item::Value(toml_edit::Value::Integer(ref expected)))
+                        if expected.value() == &0),
+                _ => true,
+            } {
+                doc["status"] = toml_edit::Item::None;
+            }
+        } else {
+            let code = code as i64;
+            match doc.get("status") {
+                Some(toml_edit::Item::Value(toml_edit::Value::String(ref expected))) => {
+                    if expected.value() != "failed" {
+                        doc["status"] = toml_edit::value("failed");
+                    }
+                }
+                Some(
+                    toml_edit::Item::Value(toml_edit::Value::InlineTable(_))
+                    | toml_edit::Item::Table(_),
+                ) => {
+                    if !matches!(
+                        doc["status"].get("code"),
+                        Some(toml_edit::Item::Value(toml_edit::Value::Integer(ref expected)))
+                            if expected.value() == &code)
+                    {
+                        doc["status"]["code"] = toml_edit::value(code);
+                    }
+                }
+                _ => {
+                    doc["status"] = toml_edit::value(toml_edit::InlineTable::default());
+                    doc["status"]["code"] = toml_edit::value(code);
+                }
+            }
+        }
+    } else if !matches!(
+        doc.get("status"),
+        Some(toml_edit::Item::Value(toml_edit::Value::String(ref expected)))
+            if expected.value() == "interrupted")
+    {
+        doc["status"] = toml_edit::value("interrupted");
+    }
+
+    Ok(doc.to_string())
+}
+
+fn overwrite_trycmd_status(
+    exit: Option<ExitStatus>,
+    expected_status: Option<CommandStatus>,
```

Comment:
Is this to save us from re-parsing the line?

How bad would it be to do that?  From a code clarify perspective, it feels weird that we are trusting to correctly pass in the correct piece of information rather than using what is already passed in

Reply:
> Is this to save us from re-parsing the line?

Yes

I can combine `expected_status` and `expected_status_source` into one parameter `step` and use `parse_trycmd` in the tests if you think it is a better way to do it

Reply:
Not looking too closely at the code, I lean towards that direction but if things end up being messier from it, I can understand.  Its just a direction I lean

Reply:
Switched to `step` and also figured out I have been missing the fences for trycmd tests the whole time